src/pages/programIndicators/form/AnalyticsPeriodBoundariesField.tsx

```diff
@@ -0,0 +1,351 @@
+    },
+]
+
+const APBItem = ({
+    apb,
+    onRemove,
+    onEdit,
+}: {
+    apb: Partial<AnalyticsPeriodBoundary>
+    onRemove: () => void
+    onEdit: (apb: Partial<AnalyticsPeriodBoundary>) => void
+}) => {
+    const foundBoundaryTarget = useMemo(
+        () =>
+            boundaryTargets.find(
+                (target) => apb.boundaryTarget === target.value
+            ),
+        [apb]
+    )
+    const foundBoundaryType = useMemo(
+        () =>
+            boundaryTypes.find(
+                (type) => apb.analyticsPeriodBoundaryType === type.value
+            ),
+        [apb]
+    )
+    return (
+        <div
+            className={classes.analyticsPeriodBoundaryContainer}
+            data-test="analytics-period-boundary"
+        >
+            <div>
+                <div>
+                    {foundBoundaryTarget?.label ??
+                        `${i18n.t('Custom')}: ${apb.boundaryTarget}`}
+                </div>
+                <div className={classes.analyticsPeriodBoundarySpan}>
+                    {apb.offsetPeriodType && (
+                        <div>
+                            {i18n.t('Period')}: {apb.offsetPeriodType}
+                        </div>
+                    )}
+                    {apb.offsetPeriods !== undefined && (
+                        <div>
+                            {i18n.t('Offset')}: {apb.offsetPeriods}
+                        </div>
+                    )}
+                    {foundBoundaryType && (
+                        <div>
+                            {i18n.t('Type')}: {foundBoundaryType.label}
+                        </div>
+                    )}
+                </div>
+            </div>
+            <ButtonStrip>
+                <Button
+                    small
+                    dataTest="apb-edit-button"
+                    onClick={() => {
+                        onEdit(apb)
+                    }}
+                >
+                    {i18n.t('Edit')}
+                </Button>
+                <Button
+                    small
+                    dataTest="apb-remove-button"
+                    secondary
+                    destructive
+                    onClick={() => {
+                        onRemove()
+                    }}
+                >
+                    {i18n.t('Remove')}
+                </Button>
+            </ButtonStrip>
+        </div>
+    )
+}
+const APBItemForm = ({
+    APB,
+    onFieldChange,
+}: {
+    APB: Partial<AnalyticsPeriodBoundary>
+    onFieldChange: (apb: Partial<AnalyticsPeriodBoundary>) => void
+}) => {
+    const [customBoundaryTarget, setCustomBoundaryTarget] = useState(false)
+    useEffect(() => {
+        const filteredBoundaryTargets = boundaryTargets.filter(
+            (e) => e.value !== 'CUSTOM' && e.value === APB.boundaryTarget
+        ).length
+        if (APB.boundaryTarget && filteredBoundaryTargets === 0) {
+            setCustomBoundaryTarget(true)
+        }
+    }, [APB])
```

Comment:
Here comes the unavoidable `useEffect` comment... 

Is `useEffect` necessary here? You could calculate this in the initial state, and use `key={editedAPB.id}` to reset the form if the boundary changed?